da_demo/src/main.rs

```diff
@@ -0,0 +1,34 @@
+#![no_main]
+#![no_std]
+
+// Needed because of the critical-section stuff not getting loaded :P
```

Comment:
this comment is unclear. Consider this instead
```suggestion
// Ensures that critical-section impls are linked against.
```